### PR TITLE
fix: check for array child enum

### PR DESCRIPTION
**Pull request by domagojk**
For stoplightio/elements#1665

I believe the issue occurs because type: array node is not recursively processed like object type so enum property was never found. This fixes the issue described in the ticket but it also changed the jest snapshot for a schema defined in `src/__fixtures__/tickets.schema.json` and indeed, found enums that should be included for a property like this:

```json
"availableDeliveryTypes": {
  "type": "array",
  "items": {
    "type": "string",
    "enum": [
      "POSTAL",
      "PICK_UP_STATION",
      "E-MAIL",
      "LOYALTY_CARD"
    ]
  }
}
```

### CODE REVIEW

File: src/components/shared/Validations.tsx
Comment (P0lip) [inline src/components/shared/Validations.tsx:209]: tgHmmm, but arrays can also have enums at the top, i.e.

```json
{
  "type": "array",
  "enum": [
    [0, 1],
    [2, 3]
  ],
  "items": {
    "type": "number"
  }
}
```

plus, there's also another "tuples" scenario possible:

```json
{
  "type": "array",
  "items": [
    {
      "type": "number",
      "enum": [0, 1, 2]
    },
    {
      "type": "string",
      "enum": ["a", "b", "c"]
    }
  ]
}
```

**Review by P0lip — COMMENTED** [review]
Overall, arrays are troublesome to handle due to the way we represent them in the UI :/
I'll the final decision to you.

File: src/components/shared/Validations.tsx
Comment (domagojk) [reply src/components/shared/Validations.tsx:209]: I knew you will find some cases I overlooked 😄 

If enum is defined for an array, validations will take take that one over child ones, so I think that solved the first thing, right?

I will check what do to with stuff like "touples", right now it wouldn't be supported since I only do this if there is a single item in array

Comment (P0lip) [reply src/components/shared/Validations.tsx:209]: > If enum is defined for an array, validations will take take that one over child ones, so I think that solves the first thing, right?

The tricky part is that we should try to display both, because both enums impose certain restrictions.

Example:

```json
{
  "type": "array",
  "enum": [
    ["Maribor", "Zagreb", "Banja Luka", "Skopje"],
    ["Warsaw", "Prague", "Budapest", "Bratislava"]
  ],
  "items": {
    "type": "string",
    "enum": ["Warsaw", "Zagreb", "Prague", "Budapest", "Bratislava", "Maribor"]
  }
}
```

Now, this is valid `["Warsaw", "Prague", "Budapest", "Bratislava"]`, and this is not valid `["Maribor", "Zagreb", "Banja Luka", "Skopje"]`.
The former is valid because all constraints imposed by 2 enums are met, while the latter is invalid because the second enum restricts a certain set of cities.

Does that make sense?

In general, there's no good way to solve this issue. 
I'm just trying to point out the potential problem we might see.

I'm quite convinced that what you implemented here will be good enough for most cases, so we might just take it and merge it.

Comment (domagojk) [reply src/components/shared/Validations.tsx:209]: Gotcha, well I would then go with current solution. It is not perfect but neither is what is in prod now. We can then revisit more complex enum scenarios later, if needed.

I'm adding @mnaumanali94 as a reviewer as well, so it is not only my decision 😄

Comment (mnaumanali94) [reply src/components/shared/Validations.tsx:209]: Agreed. Lets get this in and as customers find/report complex cases we'll look into it.

**Review by mnaumanali94 — APPROVED** [review]

**Comment by domagojk** [comment]
@P0lip having in mind that we agreed to release this non-perfect version, could you approve it? (I can't merge it now 😢 )

**Review by P0lip — APPROVED** [review]

**Pull request merged**
